### PR TITLE
Ts migrations for Menu

### PR DESCRIPTION
**Pull request by gustavo-salazar**
Migrating to TS and vs framework the components related to the menus:
* `src/components/Menu/DynamicMenu`
* `src/components/Menu/EBIMenu`
* `src/components/Menu/EntitiesMenu`
* `src/components/Menu/InterProMenu`
* `src/components/Menu/MenuItem`
* `src/components/Menu/SideMenu`
* `src/components/Menu/SideMenu/ServerStatus`
* `src/menuConfig`

Deleted because it wasn't in use anymore:
* `src/components/Menu/SingleEntityMenu`

Other fixes:
* The hamburger menu button now toggles the menu on and off
* `main` is not mandatory in `InterProPartialDescription` type. As some of the updated links here showed.

### CODE REVIEW

File: src/components/Menu/SideMenu/index.tsx [hunk 1]
```diff
@@ -73,14 +62,13 @@ export class SideMenu extends PureComponent /*:: <Props, State> */ {
           {this.props.visible && this.props.showConnectionStatusToast ? (
             <Tip
               title="💡 What is the status of external dependencies?"
```
Comment (matthiasblum) [inline src/components/Menu/SideMenu/index.tsx:64]:
```suggestion
              title="💡 What is the status of external services?"
```

File: src/components/Menu/SideMenu/index.tsx [hunk 2]
```diff
@@ -73,14 +62,13 @@ export class SideMenu extends PureComponent /*:: <Props, State> */ {
...
-              body="The green signal is displayed when an external dependency is available whist a red signal indicates
-              that InterPro cannot connect to the resource."
+              body="The green signal is displayed when an external dependency is available whist a red signal indicates that InterPro cannot connect to the resource."
```
Comment (matthiasblum) [inline src/components/Menu/SideMenu/index.tsx:65]:
```suggestion
              body="A green signal indicates that the external service is online and available. A red signal indicates that InterPro is unable to connect to the external resource."
```

**Review by matthiasblum — APPROVED** [review]
LGTM. 

> The hamburger menu button now toggles the menu on and off

You mean this one?

image

image

It wasn't already toggling the side menu on and off? :raised_eyebrow:

**Comment by gustavo-salazar** [comment]
> It wasn't already toggling the side menu on and off? 🤨

Nope, somehow was only opening it, even if the animation made it an X the action was still to open 🤷🏽

**Comment by matthiasblum** [comment]
> > It wasn't already toggling the side menu on and off? 🤨
> 
> Nope, somehow was only opening it, even if the animation made it an X the action was still to open 🤷🏽

I'll take you word for it, clicking the cross button closes the menu on my Firefox and Chromium :man_shrugging:.

**Review by typhainepl — CHANGES_REQUESTED** [review]
For some reason the `Function` right column isn't aligned anymore in the Protein page. Looking deeper in the HTML content, it looks like a div with the class `content Description_style__content___b2` has been added. Example: http://localhost:8080/interpro/protein/reviewed/O00167/

Everything else looks alright to me.

**Comment by typhainepl** [comment]
The cross wasn't closing the menu when the header was collapsed on my browser, but it was working fine in the expanded view. It's working fine now.

**Comment by typhainepl** [comment]
Not directly related to the menu TS migration, but to the use of the Visual Framework. On the Alignment subpage of Pfam pages, the `Legends` button has been changed, but not the `Downloads`. Also, the `conservation` checkbox is working fine locally, but it is disabled in the future branch. Example: http://wp-np3-bd/future/entry/pfam/PF00002/entry_alignments/?type=seed

**Comment by gustavo-salazar** [comment]
> For some reason the `Function` right column isn't aligned anymore in the Protein page. Looking deeper in the HTML content, it looks like a div with the class `content Description_style__content___b2` has been added. Example: http://localhost:8080/interpro/protein/reviewed/O00167/

Good point, that is more to do with the design changes for the `AI generated` descriptions, because we use the same component here. I will make sure to correct it.

**Comment by gustavo-salazar** [comment]
> Not directly related to the menu TS migration, but to the use of the Visual Framework. On the Alignment subpage of Pfam pages, the `Legends` button has been changed, but not the `Downloads`. 

@typhainepl  I haven't migrated the alignment page yet, however, I migrated the dropdown buttons causing the clear mismatch. I just styled that button to fix it. But unfortunately, I think we are going to see more of these cases while in the migration period. Please report them as usual, so I can correct them.

>Also, the `conservation` checkbox is working fine locally, but it is disabled in the future branch. Example: http://wp-np3-bd/future/entry/pfam/PF00002/entry_alignments/?type=seed

Damn, I thought we had fixed that one already. It is related to the way the conservation gets calculated in the background using a web worker, and how is not been deployed properly for production. I'll open an issue for it to work on it separately.

**Review by typhainepl — APPROVED** [review]
Looking good to me.

**Pull request merged**